docs/source/commands/trtllm-bench.rst

```diff
@@ -0,0 +1,141 @@
+trtllm-bench
+===========================
+
+trtllm-bench is a comprehensive benchmarking tool for TensorRT-LLM engines. It provides three main subcommands for different benchmarking scenarios:
+
+**Common Options for All Commands:**
+
+**Usage:**
+
+.. click:: tensorrt_llm.commands.bench:main
+   :prog: trtllm-bench
+   :nested: full
+   :commands: throughput, latency, build
+
+
+Dataset Preparation
+------------------
+trtllm-bench is designed to work with the [`prepare_dataset.py`](https://github.com/NVIDIA/TensorRT-LLM/blob/main/benchmarks/cpp/prepare_dataset.py) script, which generates benchmark datasets in the required format. The prepare_dataset script supports:
+
+**Dataset Types:**
+- Real datasets from various sources
+- Synthetic datasets with normal or uniform token distributions
+- LoRA task-specific datasets
+
+**Key Features:**
+- Tokenizer integration for proper text preprocessing
+- Configurable random seeds for reproducible results
+- Support for LoRA adapters and task IDs
+- Output in JSON format compatible with trtllm-bench
+
+.. important::
+   The ``--stdout`` flag is **required** when using prepare_dataset.py with trtllm-bench to ensure proper data streaming format.
+
+**prepare_dataset.py CLI Options:**
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - ``--tokenizer``
+     - Tokenizer directory or HuggingFace model name (required)
+   * - ``--output``
+     - Output JSON filename (default: preprocessed_dataset.json)
+   * - ``--stdout``
+     - Print output to stdout with JSON dataset entry on each line (**required for trtllm-bench**)
+   * - ``--random-seed``
+     - Random seed for token generation (default: 420)
+   * - ``--task-id``
+     - LoRA task ID (default: -1)
+   * - ``--rand-task-id``
+     - Random LoRA task range (two integers)
+   * - ``--lora-dir``
+     - Directory containing LoRA adapters
+   * - ``--log-level``
+     - Logging level: info or debug (default: info)
+
+**prepare_dataset.py Subcommands:**
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Subcommand
+     - Description
+   * - ``dataset``
+     - Process real datasets from various sources
+   * - ``token_norm_dist``
+     - Generate synthetic datasets with normal token distribution
+   * - ``token_unif_dist``
+     - Generate synthetic datasets with uniform token distribution
+
+**Dataset Subcommand Options:**
```

Comment:
Maybe consider using the ``dataset``/``token_norm_dist``/`token_unif_dist`` formatting in the table headings here, to make it clearer that these are related to options in the previous table. At my first read, i was slightly confused about how these tables were related.

You could also possibly create a single table with the sub commands embedded right under each command. But that would introduce more complex formatting

Reply:
Thanks for the feedback! -- I modified the layout and made it similar to what the `.. click` parsing produces. Let me know what you think!